src/hist/axestuple.py

```diff
@@ -35,6 +36,30 @@ def __getitem__(self, item: Any) -> Any:
 
         return super().__getitem__(item)
 
+    def __setattr__(self, attr: str, values: Any) -> Any:
+        if attr == "name":
```

Comment:
Can't this just be a normal `@name.setter` ? Is there something I'm missing here? I might try it just to see what happens.

Reply:
I found the reason it can't be a normal setter, and pushed that into boost-histogram. Once 1.1.1 is out, this can be a normal setter. Until then, I've adjusted the PR adapt a similar fix - it's not necessary better or worse, but will be easier to move to boost-histogram 1.1.1+ when that is released.

I've also generalized the values a bit so any arbitrary iterator can be used, and it still checks for length (also applied to boost-histogram in a slightly different way).

Reply:
(will also need to enable Python 3.10 in tests after bh 1.1.1 is out)